docs/data/sdk-middleware.md

```diff
@@ -0,0 +1,137 @@
+---
+title: Middleware
+description: Use middleware to extend Amplitude by running a sequence of custom code on every event. This pattern is flexible and you can use it to support event enrichment, transformation, filtering, routing to third-party destinations, and more.
+---
+
+Middleware lets you extend Amplitude by running a sequence of custom code on every event. This pattern is flexible and you can use it to support event enrichment, transformation, filtering, routing to third-party destinations, and more.
+
+!!!note
+    Middleware is only supported in maintenance SDK except the maintenance Browser SDK and legacy Ampli. Middleware has been replaced by **[Plugins](../sdk-plugins/)** in the latest SDK and Ampli versions.
+
+## Middleware Structure
+ 
+### Function Signature 
+
+Each middleware is a simple function with this signature:
+
+```js
+/**
+ * A function to run on the Event stream (each logEvent call)
+ *
+ * @param payload The `payload` contains the `event` to send and an optional `extra` that lets you pass custom data to your own middleware implementations.
+ * @param next Function to run the next middleware in the chain, not calling next will end the middleware chain
+ */
+
+function (payload: MiddlewarePayload: next: MiddlewareNext): void;
+```
+
+???types "Types in  Middleware"
+    |<div class="med-column">Name</div>| Type | Description |
+    | - | - | - |
+    | `MiddlewarePayload.event` | Event | The event data being sent. The event may vary by platform. |
+    | `MiddlewarePayload.extra` | { [x: string]: any } | Unstructured object to let users pass extra data to middleware. |
+    | `MiddlewareNext` | (payload: MiddlewarePayload) => void | Function called at the end of each Middleware to run the next middleware in the chain. |
+
+To invoke the next middleware in the queue, use the `next` function. You must call `next(payload)` to continue the middleware chain. If a middleware doesn't call `next`, then the event processing stops executing after the current middleware completes.
+
+### Payload Customization
+
+Middleware access to event fields may vary by platform. To ensure comprehensive access, we recommend updating to the latest Ampli version and utilizing the [Plugins](../sdk-plugins) feature.
+
+For browser ampli, the following are the accessable keys under `payload`.
+
+|<div class="med-column">Name</div>|Type|
```

Comment:
I think it would be good to add descriptions for each of these. 1) to match the other config tables 2) better to be overly explicit